torch/csrc/jit/autodiff.cpp

```diff
@@ -17,14 +17,25 @@ bool isDifferentiable(Node * n) {
   static std::unordered_set<Symbol> differentiable_kinds = {
     aten::add, aten::sub, aten::mul, prim::Constant, prim::ReplaceIfUndef,
     aten::sigmoid, aten::tanh, aten::mm, aten::chunk, aten::split, aten::t, aten::neg,
-    aten::unsqueeze
+    aten::unsqueeze, aten::expand
   };
   return differentiable_kinds.count(n->kind()) > 0;
 }
 
+static bool isDifferentiable(Block * b) {
+  for(Node*n : b->nodes()) {
+    if(!isDifferentiable(n))
+      return false;
+    for(Block* b : n->blocks()) {
+      if(!isDifferentiable(b))
+        return false;
+    }
+  }
+  return true;
+}
```

Comment:
This change seems to be quite unnecessary. I don't think we were planning to implement differentiation of control flow any time soon, so we don't really need to traverse the blocks, since `isDifferentiable(n)` will fail for any nodes that might have them.